cot/src/response/into_response.rs

```diff
@@ -0,0 +1,628 @@
+        match self {
+            Ok(value) => value.into_response(),
+            Err(err) => Err(err.into()),
+        }
+    }
+}
+
+impl IntoResponse for Response {
+    fn into_response(self) -> cot::Result<Response> {
+        Ok(self)
+    }
+}
+
+// Text implementations
+
+impl_into_response_for_type_and_mime!(&'static str, mime::TEXT_PLAIN_UTF_8);
+impl_into_response_for_type_and_mime!(String, mime::TEXT_PLAIN_UTF_8);
+
+impl IntoResponse for Box<str> {
+    fn into_response(self) -> cot::Result<Response> {
+        String::from(self).into_response()
+    }
+}
+
+// Bytes implementations
+
+impl_into_response_for_type_and_mime!(&'static [u8], mime::APPLICATION_OCTET_STREAM);
+impl_into_response_for_type_and_mime!(Vec<u8>, mime::APPLICATION_OCTET_STREAM);
+impl_into_response_for_type_and_mime!(Bytes, mime::APPLICATION_OCTET_STREAM);
+
+impl<const N: usize> IntoResponse for &'static [u8; N] {
+    fn into_response(self) -> cot::Result<Response> {
+        self.as_slice().into_response()
+    }
+}
+
+impl<const N: usize> IntoResponse for [u8; N] {
+    fn into_response(self) -> cot::Result<Response> {
+        self.to_vec().into_response()
+    }
+}
+
+impl IntoResponse for Box<[u8]> {
+    fn into_response(self) -> cot::Result<Response> {
+        Vec::from(self).into_response()
+    }
+}
+
+impl IntoResponse for BytesMut {
+    fn into_response(self) -> cot::Result<Response> {
+        self.freeze().into_response()
+    }
+}
+
+// HTTP structures for common uses
+
+impl IntoResponse for StatusCode {
+    fn into_response(self) -> cot::Result<Response> {
+        ().into_response().with_status(self)
+    }
+}
+
+impl IntoResponse for http::HeaderMap {
+    fn into_response(self) -> cot::Result<Response> {
+        ().into_response().map(|mut resp| {
+            *resp.headers_mut() = self;
+            resp
+        })
+    }
+}
+
+impl IntoResponse for http::Extensions {
+    fn into_response(self) -> cot::Result<Response> {
+        ().into_response().map(|mut resp| {
+            *resp.extensions_mut() = self;
+            resp
+        })
+    }
+}
+
+impl IntoResponse for http::response::Parts {
+    fn into_response(self) -> cot::Result<Response> {
+        Ok(Response::from_parts(self, Body::empty()))
+    }
+}
+
+// Data type structures implementations
+
+impl IntoResponse for Html {
+    /// Create a new HTML response.
+    ///
+    /// This creates a new [`Response`] object with a content type of
+    /// `text/html; charset=utf-8` and given status code and body.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::html::Html;
+    /// use cot::response::IntoResponse;
+    ///
+    /// let html = Html::new("<div>Hello</div>");
+    ///
+    /// let response = html.into_response();
+    /// ```
+    fn into_response(self) -> cot::Result<Response> {
+        self.as_str()
+            .to_owned()
+            .into_response()
+            .with_content_type(mime::TEXT_HTML_UTF_8.as_ref())
```

Comment:
We already have `crate::headers::HTML_CONTENT_TYPE`, so I think we should use it here.